### PR TITLE
Add MAGIC as Collateral

### PR DESCRIPTION
**Pull request by MrBahama1**

### CODE REVIEW

**Comment by MrBahama1** [comment]
@mrice32 Great to see you and the rest of the UMA team!!! <3



> Great to see Matt and the rest of the team!! <3

File: UMIPs/umip-162.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+## Headers
+
```
Comment (mhairimcalpine) [inline UMIPs/umip-162.md:2]: Could these be separated out into two PRs?

Comment (Reinis-FRP) [reply UMIPs/umip-162.md:2]: Also, please change file names not to include UMIP numbers as thos get assigned when merging before vote.

File: UMIPs/umip-162.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - MAGIC/USDT: OKX
+
+    https://www.coingecko.com/en/coins/magic#markets
+
+- Real-time data update frequency:
+    - AMM pools: updated every block mined
+
+# Price feed implementation
+
+This price identifier uses the XXXXX
```
Comment (Reinis-FRP) [inline UMIPs/umip-162.md:50]: Please add links to used price feeds and their configuration

File: UMIPs/umip-162.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+1. Query CVX/WETH price from Uniswap v3 and SushiSwap using 5-minute TWAP.
+2. Query the WETH/USD price as per UMIP-6.
+3. Multiply each of CVX/WETH prices in step 1 with ETH/USD price from step 2.
+4. Take the open CVX/USDT price of the 1 minute OHLC period that the timestamp falls in from Okex.
+5. Take the median of all results from step 3 and 4.
+6. Round result from step 5 to 8 decimals to get the CVXUSD price.
+7. (for USDCVX) Take the inverse of the result of step 5.
+8. (for USDCVX) Round result from step 7 to 8 decimals to get the USDCVX price.
```
Comment (Reinis-FRP) [inline UMIPs/umip-162.md:83]: This is for CVX, please update with specific instructions for MAGIC that are consistent with price feed configuration in the Price feed implementation section.

File: UMIPs/umip-162.md (hunk 2 above)
Comment (MrBahama1) [reply UMIPs/umip-162.md:50]: Will need support from DevX to create these price feeds @smb2796

**Review by therealhexi — APPROVED** [review]
Approve

File: UMIPs/MAGIC-as-Collateral.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+## Technical Specification
+
+To accomplish this upgrade, the following changes need to be made:
+
+
+- MAGIC token address 0x539bdE0d7Dbd336b79148AA742883198BBF60342
+
+https://arbiscan.io/token/0x539bdE0d7Dbd336b79148AA742883198BBF60342
+
+- A final fee of 2500 MAGIC needs to be set in the Store contract for MAGIC token
```
Comment (Reinis-FRP) [inline UMIPs/MAGIC-as-Collateral.md:26]: Please update final fee to be closer to $1500. Currently 5500 MAGIC seems appropriate.

File: UMIPs/MAGIC-as-Collateral.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+## Motivation
+
+MAGIC token is economic infrastructure for various play-to-own games using TreasureDAO. Adding MAGIC to UMA contracts will enable the Treasure ecosystem to incorporate more flexible in-game assets with defi. 
+
+## Technical Specification
+
+To accomplish this upgrade, the following changes need to be made:
+
+
+- MAGIC token address 0x539bdE0d7Dbd336b79148AA742883198BBF60342
```
Comment (Reinis-FRP) [inline UMIPs/MAGIC-as-Collateral.md:22]: Please explicitly state that this address on Arbitrum needs to be added to the collateral currency whitelist introduced in UMIP-8.
Is it only on Arbitrum?

File: UMIPs/MAGIC-as-Collateral.md (hunk 4 above)
Comment (MrBahama1) [reply UMIPs/MAGIC-as-Collateral.md]: Updated!

File: UMIPs/MAGIC-as-Collateral.md (hunk 5 above)
Comment (MrBahama1) [reply UMIPs/MAGIC-as-Collateral.md:22]: Arbitrum and Ethereum

Ethereum: 0xB0c7a3Ba49C7a6EaBa6cD4a96C55a1391070Ac9A

**Pull request merged**